### PR TITLE
add option to support customized elmo embeddings

### PR DESCRIPTION
**Pull request by GiliGoldin**
I added two optional arguments to ELMoEmbeddings's init function: 
1) options_file
2) weights_file
This way a customized ELMO model can be provided by the user for embeddings.
This is useful for using a self-trained ELMO model or any known ELMO models that are not yet supported by Flair.

### CODE REVIEW

File: flair/embeddings.py [hunk 1]
```diff
@@ -354,23 +354,23 @@ def __init__(self, model: str = 'original'):
...
-        if model == 'medium':
-            options_file = 'https://s3-us-west-2.amazonaws.com/allennlp/models/elmo/2x2048_256_2048cnn_1xhighway/elmo_2x2048_256_2048cnn_1xhighway_options.json'
-            weight_file = 'https://s3-us-west-2.amazonaws.com/allennlp/models/elmo/2x2048_256_2048cnn_1xhighway/elmo_2x2048_256_2048cnn_1xhighway_weights.hdf5'
-        if model == 'pt' or model == 'portuguese':
-            options_file = 'https://s3-us-west-2.amazonaws.com/allennlp/models/elmo/contributed/pt/elmo_pt_options.json'
-            weight_file = 'https://s3-us-west-2.amazonaws.com/allennlp/models/elmo/contributed/pt/elmo_pt_weights.hdf5'
-        if model == 'pubmed':
-            options_file = 'https://s3-us-west-2.amazonaws.com/allennlp/models/elmo/contributed/pubmed/elmo_2x4096_512_2048cnn_2xhighway_options.json'
-            weight_file = 'https://s3-us-west-2.amazonaws.com/allennlp/models/elmo/contributed/pubmed/elmo_2x4096_512_2048cnn_2xhighway_weights_PubMed_only.hdf5'
+        if not options_file or not weight_file:
```
Comment (stefan-it) [inline flair/embeddings.py]: I think we can just initialize both `options_file` and `weight_file` via constructor argument:

```python
def __init__(self, model: str = 'original', 
                    options_file: str = allennlp.commands.elmo.DEFAULT_OPTIONS_FILE,
                    weight_file: str = allennlp.commands.elmo.DEFAULT_WEIGHT_FILE):
```

Then it is semantically clear, that these default values are used + no `if not` checks are needed here :)

**Comment by stefan-it** [comment]
Thanks for that PR :) I just left a review ☺️

File: flair/embeddings.py (hunk 1 above)
Comment (GiliGoldin) [reply flair/embeddings.py]: @stefan-it 
Hi, 
thanks for the review.

This is what i initially wanted to do, but the code currently imports allennlp only at a later stage, in a try except block
```python
try:
      import allennlp.commands.elmo
except:
```
so the default options and weight files are unavailable in the constructor.
Any suggestions?

Comment (stefan-it) [reply flair/embeddings.py]: Oh, sorry. I forgot that we import `allennlp` later...

**Comment by alanakbik** [comment]
:+1:

**Comment by alanakbik** [comment]
@GiliGoldin thanks for adding this! 
@stefan-it I've merged to the new code formatting style - could you +1 and merge?

**Comment by stefan-it** [comment]
:+1:

**Comment by alanakbik** [comment]
Thanks!

**Pull request merged**